File: 2395-find-subarrays-with-equal-sum/2395-find-subarrays-with-equal-sum.rb
# @param {Integer[]} nums
# @return {Boolean}
def find_subarrays(nums)
    res=[]
    (nums.size-1).times do |i|
        windowsum = nums[i..i+1].sum
        return true if res.include?(windowsum)
        res << windowsum
    end
    return false
end

File: 2395-find-subarrays-with-equal-sum/2395-find-subarrays-with-equal-sum.py
class Solution:
    def findSubarrays(self, nums: List[int]) -> bool:
        res=[]
        for i in range(len(nums)-1):
            windowsum = sum(nums[i:i+2])
            if windowsum in res: 
                return True 
            res.append(windowsum)

        return False        